mobile/lib/features/trade/position_change_notifier.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:collection';
+import 'dart:developer';
+import 'package:flutter/material.dart';
+import 'package:get_10101/common/application/event_service.dart';
+import 'package:get_10101/features/trade/application/position_service.dart';
+import 'package:get_10101/features/trade/domain/contract_symbol.dart';
+import 'package:get_10101/bridge_generated/bridge_definitions.dart' as bridge;
+
+import 'domain/position.dart';
+
+class PositionChangeNotifier extends ChangeNotifier implements Subscriber {
+  HashMap<ContractSymbol, Position> positions = HashMap();
+
+  Future<void> _create(PositionService positionService) async {
+    List<Position> positions = await positionService.fetchPositions();
+    for (Position position in positions) {
+      this.positions[position.contractSymbol] = position;
```

Comment:
Is it worth checking that there are no duplicate positioned returned by the position service? Maybe just in debug. Unless this is actually allowed?

Reply:
I would not bother for now - I would see such a check more on the database level. If we implement the data model right we should not have this problem. 
Checking on the consumer side for sanity can get out of hand quite fast.

One thing that's a bit neglected at the moment is tests - I would see most unit tests on the DB and business logic layers though; so far it is mostly UI and glue code.